pkg/serviceaccounts/controllers/mutation_cache.go

```diff
@@ -0,0 +1,88 @@
+package controllers
+
+import (
+	"github.com/golang/glog"
+	lru "github.com/hashicorp/golang-lru"
+
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage/etcd"
+)
+
+// MutationCache is able to take the result of update operations and stores them in an LRU
+// that can be used to provide a more current view of a requested object.  It requires interpretting
```

Comment:
> Is this "best effort" or "guaranteed" to see more recent mutations? Single threaded access or thread safe?

LRU is threadsafe, so this is threadsafe.  It is best effort, since LRU can page out.